store/postgres/src/subgraph_store.rs

```diff
@@ -997,10 +1005,25 @@ impl SubgraphStoreTrait for SubgraphStore {
         Ok(info.api)
     }
 
-    fn debug_fork(&self, id: &DeploymentHash) -> Result<Option<DeploymentHash>, StoreError> {
-        let (store, site) = self.store(&id)?;
+    fn debug_fork(
+        &self,
+        id: &DeploymentHash,
+        logger: Logger,
+    ) -> Result<Option<Arc<dyn SubgraphFork>>, StoreError> {
+        let (store, site) = self.store(id)?;
         let info = store.subgraph_info(&site)?;
-        Ok(info.debug_fork)
+        let fork_id = info.debug_fork;
+        let schema = self.input_schema(id)?;
+
+        match (self.fork_base.as_ref(), fork_id) {
+            (Some(b), Some(f)) => Ok(Some(Arc::new(fork::SubgraphFork::new(
+                b.join(&format!("id/{}", f))
```

Comment:
The reason this used to fail with URLs that ended up with no `/` is because `.join` **replaces** the "file" from the path if it's not a folder, eg:

```rust
// Without `/`
let base = Url::parse("https://api.thegraph.com/subgraphs/id")?;
let url = base.join("QmSH3hhL9vAxQbE6ydM6eYN9ULtf6WS7hCHgQqxAhMJwjL")?;
// Removed the `/id`
assert_eq!(url.as_str(), "https://api.thegraph.com/subgraphs/QmSH3hhL9vAxQbE6ydM6eYN9ULtf6WS7hCHgQqxAhMJwjL");

// With `/`
et base = Url::parse("https://api.thegraph.com/subgraphs/id/")?;
let url = base.join("QmSH3hhL9vAxQbE6ydM6eYN9ULtf6WS7hCHgQqxAhMJwjL")?;
// Kept `/id`
assert_eq!(url.as_str(), "https://api.thegraph.com/subgraphs/id/QmSH3hhL9vAxQbE6ydM6eYN9ULtf6WS7hCHgQqxAhMJwjL");
```

`Url::join` docs: https://docs.rs/url/latest/url/struct.Url.html#method.join